python/sqlflow_submitter/tensorflow/train.py

```diff
@@ -122,37 +121,34 @@ def train_input_fn():
         train_dataset = train_dataset.shuffle(SHUFFLE_SIZE).batch(batch_size).cache().repeat(epochs if epochs else 1)
         return train_dataset
 
-    if validate_select == "":
-        classifier.train(input_fn=lambda:train_dataset)
```

Comment:
The bug here is never triggered because the control flow never passes the condition check.

Reply:
I think the fix should be: do not add default validation select, and if validation select is empty, do not run the validation.

Reply:
The aim of the commit is to remove useless(and buggy) code, which does not change any logic. What you suggest is another topic that slightly changes users' experience. How about addressing these two requirements separately? The latter can be provided in another pull request after some discussion.

Reply:
If we agree with "do not add default validation select", then some changes made in this PR need to be reverted once we merged this PR. I recommend close this PR and file another PR to directly enhance this feature.